src/rust/src/lib.rs

```diff
@@ -83,6 +83,10 @@ fn _legacy_provider_error(success: bool) -> pyo3::PyResult<()> {
     Ok(())
 }
 
+fn _str_ref_to_cstr_ref(message: &str) -> &std::ffi::CStr {
```

Comment:
I could also have done it without adding this helper to the crate root. Happy to do it a more idiomatic way or just drop the helper if you don't think it's worth it.